java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure2.ql

```diff
@@ -0,0 +1,72 @@
+/**
+ * @name Temporary Directory Local information disclosure
+ * @description Detect local information disclosure via the java temporary directory
+ * @kind path-problem
+ * @problem.severity warning
+ * @precision very-high
+ * @id java/local-information-disclosure
+ * @tags security
+ *       external/cwe/cwe-200
+ */
+
+import TempDirUtils
+import DataFlow::PathGraph
+
+private class MethodFileSystemFileCreation extends Method {
+  MethodFileSystemFileCreation() {
+    getDeclaringType() instanceof TypeFile and
+    (
+      hasName("mkdir") or
+      hasName("createNewFile")
+    )
+  }
+}
+
+private class MethodFilesSystemFileCreation extends Method {
+  MethodFilesSystemFileCreation() {
+    getDeclaringType().hasQualifiedName("java.nio.file", "Files") and
+    hasName("write")
```

Comment:
There are many more file creation methods in the `Files` class, `newBufferedWriter`, `newOutputStream`, `newByteChannel`, `createFile`, `createDirectory`, `createDirectories` and perhaps more.

Reply:
Have all of these been captured into a common predicate yet?

Reply:
I don't think so, a `git grep createDirectory` has no results.

Reply:
So, I just tried `newByteChannel`, it looks like even though the documentation states that it will create a file if none exists, that doesn't look like it's accurate. Not sure if you have other experiences here.